File: 907-koko-eating-bananas/koko-eating-bananas.java
class Solution {
    public long helper(int[] piles,int k){
        long res = 0;
        for(int i:piles){
            res += (int) Math.ceil((double) i / k);
        }
        return res;
    }
    public int minEatingSpeed(int[] piles, int h) {
        int low = 1;
        int high = 0;
        for(int i:piles){
            high = Math.max(i,high);
        }
        int ans = 0;
        while(low<=high){
            int mid = low+(high-low)/2;
            long res = helper(piles,mid);
            System.out.println(res);
            if(res<=h){
                ans = mid;
                high = mid-1;
            }
            else low = mid+1;
        }
        return ans;
    }
}

File: 907-koko-eating-bananas/koko-eating-bananas.cpp
class Solution {
public:
    long long helper(vector<int>& piles,int k){
        long long res = 0;
        for(int i:piles){
            if(i%k==0) res+=i/k;
            else res+=(i/k)+1;
        }
        return res;
    }
    int minEatingSpeed(vector<int>& piles, int h) {
        int l = 1;
        int r = *max_element(piles.begin(),piles.end());
        while(l<=r){
            int mid = l + (r - l) / 2;
            if(helper(piles,mid)<=h){
                r = mid-1;
            } else {
                l = mid+1;
            }
        }
        return l;
    }
};